spacy/util.py

```diff
@@ -738,6 +738,24 @@ def get_package_path(name: str) -> Path:
     return Path(pkg.__file__).parent
 
 
+def replace_model_node(model: Model, target: Model, replacement: Model) -> None:
```

Comment:
Arguably we should have this in Thinc. If we do put a version of this in Thinc I guess we can remove this utility later.

Reply:
I'm not really convinced that problem needs solving. Sure, we'll have a ref there named listener that returns a non-listener result, due to this quirk of the model's history. But it does return the thing the listener pointed to, so...Trying to clean it up would potentially cause just as many, but different, problems.

Reply:
It's not a huge problem and it won't lead to any bugs *right now*, but I can just feel us coming back to this in 6 months going "oh right, that listener ref is not actually a listener and now this new thing here broke down". I'm OK with leaving this as is though, because my proposed solution definitely isn't clean either.